### PR TITLE
Bugfix/auth fix

### PR DESCRIPTION
**Pull request by hmeister10**

### CODE REVIEW

File: api/auth.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const guestLogin = async (name: string): Promise<User> => {
 };
 
 export const isAuthenticated = () => {
-  return !!auth.getToken() && auth.getToken() != "";
+  return !!auth.getToken();
```
Comment (neha-gehani) [inline api/auth.ts:43]: @hmeister10 Did we figure out why this was returning true for jatin?

Comment (neha-gehani) [reply api/auth.ts:43]: Also look at the constructor, undefined token will still get set no?

File: components/global/AuthenticatedRoute.tsx [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+  const fetchUser = async () => {
+    const userData = await getUser();
+    if(!userData){
+      Router.push('/login');
+    }
+    setHasFetchedUser(false);
+    setUser(userData);
+  };
+
+  const [hasFetchedUser, setHasFetchedUser] = useState(true);
```
Comment (neha-gehani) [inline components/global/AuthenticatedRoute.tsx:29]: Move this up, its defined after its used. Not an error but still

File: components/global/AuthenticatedRoute.tsx [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
+
+  useEffect(() => {
+    if(hasFetchedUser) {
+      const isLoggedIn = isAuthenticated();
+
+      if(!isLoggedIn) {
+        Router.push('/login');
+      } else {
+        console.log('Fetched user')
+        fetchUser();
```
Comment (neha-gehani) [inline components/global/AuthenticatedRoute.tsx:39]: why?

**Review by neha-gehani — CHANGES_REQUESTED** [review]

File: api/auth.ts (hunk 1 above)
Comment (hhiranandani) [reply api/auth.ts:43]: no the constructor also calls the `setToken` method, which checks before setting the cookie

File: components/global/AuthenticatedRoute.tsx (hunk 2 above)
Comment (hhiranandani) [reply components/global/AuthenticatedRoute.tsx:29]: its not used before defined, the functions are defined
the useEffect is after this

Comment (neha-gehani) [reply components/global/AuthenticatedRoute.tsx:29]: In code i mean. `setHasFetchedUser` is used on line 25

**Review by neha-gehani — APPROVED** [review]

**Pull request merged**